packages/sw-precaching/src/lib/revisioned-cache-manager.js

```diff
@@ -158,23 +179,43 @@ class RevisionedCacheManager {
     cacheName = cacheName || defaultCacheName;
 
     let openCache = await caches.open(cacheName);
-    const cachePromises = this._fileEntriesToCache.map(async (fileEntry) => {
+    const fileEntriesToCache = Object.values(this._fileEntriesToCache);
+    const cachePromises = fileEntriesToCache.map(async (fileEntry) => {
       const isCached = await this._isAlreadyCached(fileEntry, openCache);
       if (isCached) {
         return;
       }
 
       let requestUrl = this._cacheBustUrl(fileEntry);
-      const response = await fetch(requestUrl, {
+      let fetchOptions = {
         credentials: 'same-origin',
-      });
-      await openCache.put(fileEntry.path, response);
-      await this._putRevisionDetails(fileEntry.path, fileEntry.revision);
+      };
+
+      if (requestUrl.indexOf(location.origin) !== 0) {
```

Comment:
As per the earlier conversation, I'd avoid doing this implicit `no-cors` fallback.